packages/util/test/bytes.spec.ts

```diff
@@ -372,6 +384,26 @@ describe('bytesToBigInt', () => {
     const buf = hexToBytes('0x123')
     assert.strictEqual(BigInt(0x123), bytesToBigInt(buf))
   })
+  it('should return 0n for empty Uint8Array', () => {
+    assert.strictEqual(bytesToBigInt(new Uint8Array(0)), 0n)
+  })
+  it('should throw if input is not a Uint8Array', () => {
+    assert.throws(function () {
+      // @ts-expect-error -- Testing invalid input
+      bytesToBigInt([1, 2, 3])
+    }, /This method only supports Uint8Array/)
+  })
+  it('should not mutate input when littleEndian is true', () => {
+    const bytes = new Uint8Array([0x01, 0x02, 0x03])
+    const original = new Uint8Array([0x01, 0x02, 0x03])
+    bytesToBigInt(bytes, true)
+    assert.deepEqual(bytes, original)
```

Comment:
What does this test? I'm not sure why this would mutate, is this because of the little endianness? So we would "mirror" the array so we can convert it and we thus check that it does not alter the original?